### DISCUSSION TITLE
(How) Can I rearrange groups in custom order?

### DISCUSSION THREAD

**Discussion by grossardt** [keepassxreboot/keepassxc#9890] [Q&A]
I am trying to get the groups in the left pane of my database into custom order but fail in two different settings (2.6.6@Ubuntu, 2.7.5@Windows). Am I missing something, or is there no way to rearrange groups? I am quite certain this was possible in keepass. Yet drag&drop has no effect in either setup (I can move a group into another but not rearrange the order).

Are there any complications speaking against adding "move up/down" context menu options as a feature as they exist for entries?

Closest I could find was a suggested workaround to move everything into a temp folder and then back out in the preferred order, but this is obviously a huge annoyance and barely a solution. Yet, there seems no open bug/feature request addressing this issue?

**Comment by droidmonkey** [comment]
Drag and drop is the answer. You have to line up the drop between groups, you should see a line appear.

**Reply by grossardt** [reply]
I see the line but nothing is being moved. Two different databases on two different systems with two different keepass xc versions.

**Reply by droidmonkey** [reply]
![2023-10-02_07-10-21](https://github.com/keepassxreboot/keepassxc/assets/2809491/23fbafeb-c10c-4c22-ae21-737cd931826e)

**Reply by grossardt** [reply]
It gets weirder: On the setup with 2.6.6 / Linux I created a new database for testing. Not only could I move groups in the new database, after closing and reopening, it now also works in the old database.
**However**, trying the exact same method on the 2.7.5 / Windows computer did not work: here I can **also** move groups in a new database, but even after closing and reopening it still does not work for the old database.

**Reply by droidmonkey** [reply]
That is weird because there is nothing like a setting or whatever that would prevent moving groups.
